src/components/reusable/blockCodeView/blockCodeView.ts

```diff
@@ -0,0 +1,520 @@
+import { html, LitElement } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+
+import Prism from 'prismjs';
+import 'prismjs/plugins/autoloader/prism-autoloader';
+import 'prismjs-components-importer';
+Prism.plugins.autoloader.languages_path = 'node_modules/prismjs/components/';
+
+import copyIcon from '@carbon/icons/es/copy/20';
+import checkmarkIcon from '@carbon/icons/es/checkmark--outline/20';
+import chevronDown from '@carbon/icons/es/chevron--down/20';
+
+import '@kyndryl-design-system/shidoka-foundation/components/button';
+import '@kyndryl-design-system/shidoka-foundation/components/icon';
+
+import BlockCodeViewStyles from './blockCodeView.scss';
+import ShidokaLightTheme from './shidokaLightSyntaxStyles.scss';
+import ShidokaDarkTheme from './shidokaDarkSyntaxStyles.scss';
+
+interface LanguageMatch {
+  language: string;
+  relevance: number;
+}
+
+/**
+ * `<kyn-block-code-view>` component to display `<code>` snippets as standalone single-/multi-line block elements.
+ * @fires on-custom-copy - Emits when the copy button is clicked.
```

Comment:
`on-copy` is probably fine. Using the word `custom` for a custom event name is a bit redundant.

Reply:
✅